server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/views/agents/environments_list_widget.js.msx

```diff
@@ -0,0 +1,106 @@
+define(['mithril', 'lodash', 'jquery', 'js-routes', 'helpers/mrequest',
+    'helpers/form_helper', 'helpers/mithril_component_mixins', 'views/agents/environment_checkbox_widget'],
+  function (m, _, $, Routes, mrequest, f, ComponentMixins,
+            EnvironmentCheckboxWidget) {
+    var foundationReflow = function () {
+      return function (elem, isInitialized) {
+        if (!isInitialized) {
+          new Foundation.Dropdown($(elem));
+        }
+      };
+    };
+
+    var EnvironmentsListWidget = {
+      controller: function (args) {
+        var self = this;
+
+        this.selectedAgentsEnvironments = function () {
+          var checkedAgents = _.compact(args.selectAllViewModel.checkedAgents());
+          return _.reduce(checkedAgents, function (environments, agent) {
+            environments[agent.hostname()] = agent.environments();
+            return environments;
+          }, {});
+        };
+
+        this.updateEnvironments = function () {
+          args.togglePolling(false);
+          $('#environments_list').removeClass('is-open');
```

Comment:
Please don't do this. You're mutating virtual dom element from another using jQuery. Ideally you'd want the `#environments_list` element's `class` to be set via a function that's controlled by a flag.